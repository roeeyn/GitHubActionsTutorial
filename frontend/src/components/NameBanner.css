.wrapper {
  margin: 0;
}

.banner {
  margin: inherit;
  font-size: 6em;
}
