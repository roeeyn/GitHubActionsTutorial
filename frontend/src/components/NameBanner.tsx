import React from "react";
import "./NameBanner.css";

interface NameBannerProps {
  name?: string;
}

export const NameBanner = ({ name = "World" }: NameBannerProps) => {
  return (
    <section className="wrapper">
      <p className="banner">Hello {name !== "" ? name : "World"}!</p>
    </section>
  );
};
