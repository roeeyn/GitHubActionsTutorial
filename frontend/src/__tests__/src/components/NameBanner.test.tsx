import React from "react";
import renderer from "react-test-renderer";
import { NameBanner } from "../../../components/NameBanner";

it("Renders Hello World", () => {
  const tree = renderer.create(<NameBanner name={""} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("Renders Pedrito", () => {
  const tree = renderer.create(<NameBanner name={"Pedrito"} />).toJSON();
  expect(tree).toMatchSnapshot();
});
