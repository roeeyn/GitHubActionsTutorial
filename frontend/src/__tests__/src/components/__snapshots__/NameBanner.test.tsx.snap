// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Hello World 1`] = `
<section
  className="wrapper"
>
  <p
    className="banner"
  >
    Hello
    World
    !
  </p>
</section>
`;

exports[`Renders Pedrito 1`] = `
<section
  className="wrapper"
>
  <p
    className="banner"
  >
    Hello
    Pedrito
    !
  </p>
</section>
`;
